uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/browser/BrowserEdge.java

```diff
@@ -155,17 +145,28 @@ private void signInWithWorkOrSchoolAccount(@NonNull final String username,
                                                @NonNull final String password,
                                                @NonNull final FirstPartyAppPromptHandlerParameters promptHandlerParameters) throws UiObjectNotFoundException {
         final UiObject signInWithWorkAccountBtn = UiAutomatorUtils.obtainUiObjectWithText(
-                "Sign in with a work or school account"
+                "Add account"
         );
 
         // click Sign In with work or school account btn
         signInWithWorkAccountBtn.click();
 
         Logger.i(TAG, "Handle Sign-In Prompt for Work or School account..");
-        // handle prompt
-        final AadPromptHandler aadPromptHandler = new AadPromptHandler(promptHandlerParameters);
-        aadPromptHandler.handlePrompt(username, password);
+        // handle email field
+        final UiObject emailField = UiAutomatorUtils.obtainUiObjectWithUiSelector(new UiSelector().className("android.widget.EditText"), CommonUtils.FIND_UI_ELEMENT_TIMEOUT);
+        try {
+            emailField.setText(username);
+            UiAutomatorUtils.handleButtonClickForObjectWithText("Next");
+        }catch(UiObjectNotFoundException ex){
+            throw new AssertionError(ex);
+        }
+
+        //handle password field
+        Logger.i(TAG, "Handle Aad Login Password UI..");
+        UiAutomatorUtils.handleInput("i0118", password);
+        UiAutomatorUtils.handleButtonClick("idSIButton9");
```

Comment:
I believe this is a deviation from what we agreed on during our call. We should still use these lines:
        final AadPromptHandler aadPromptHandler = new AadPromptHandler(promptHandlerParameters);
        aadPromptHandler.handlePrompt(username, password);

This is so that we handle any other behavior that comes up for an account (something like "Stay Signed In" page or consent page would not be handled in this operation), all of which is contained in the AadPromptHandler call.

Reply:
I see you've added the username as loginHint in your broker PR, so that is good (this means the AadPromptHandler will not try to handle email page, since it's already being handled earlier in this method.